kolibri/tasks/api.py

```diff
@@ -143,68 +169,103 @@ def localdrive(self, request):
         return Response(out)
 
 
-def _networkimport(channel_id, update_progress=None):
+def _networkimport(channel_id, update_progress=None, check_for_cancel=None):
     call_command(
         "importchannel",
         "network",
         channel_id,
-        update_progress=update_progress)
-    call_command(
-        "importcontent",
-        "network",
-        channel_id,
-        update_progress=update_progress)
+        update_progress=update_progress,
+        check_for_cancel=check_for_cancel)
+    try:
+        call_command(
+            "importcontent",
+            "network",
+            channel_id,
+            update_progress=update_progress,
+            check_for_cancel=check_for_cancel)
+    except UserCancelledError:
+        connections.close_all()  # close all DB connections (FIX for #1818)
+        try:
+            os.remove(get_content_database_file_path(channel_id))
+        except OSError:
+            pass
+        ChannelMetadataCache.objects.filter(id=channel_id).delete()
+        raise
     connections.close_all()  # close all DB connections (FIX for #1818)
 
-def _localimport(drive_id, update_progress=None):
+def _localimport(drive_id, update_progress=None, check_for_cancel=None):
     drives = get_mounted_drives_with_channel_info()
     drive = drives[drive_id]
-    # copy channel's db file then copy all the content files from sorage dir
-    for channel in drive.metadata["channels"]:
-        call_command(
-            "importchannel",
-            "local",
-            channel["id"],
-            drive.datafolder,
-            update_progress=update_progress)
-        call_command(
-            "importcontent",
-            "local",
-            channel["id"],
-            drive.datafolder,
-            update_progress=update_progress)
+    # copy channel's db file then copy all the content files from storage dir
+    try:
+        for channel in drive.metadata["channels"]:
+            call_command(
+                "importchannel",
+                "local",
+                channel["id"],
+                drive.datafolder,
+                update_progress=update_progress,
+                check_for_cancel=check_for_cancel)
+            call_command(
+                "importcontent",
+                "local",
+                channel["id"],
+                drive.datafolder,
+                update_progress=update_progress,
+                check_for_cancel=check_for_cancel)
+    except UserCancelledError:
+        connections.close_all()  # close all DB connections (FIX for #1818)
+        for channel in drive.metadata["channels"]:
+            channel_id = channel["id"]
```

Comment:
I'd move the `connections.close_all()` right after this line to be sure. Or add another one... basicaully juyst keep this text in your clipboard and when in doubt paste it here.